Wine Brand prediction

This model is to predict the wine brand from the metrics that is provided in the dataset.

In [1]:
#1
import pandas as pd

In [2]:
#2
wine = pd.read_csv('https://github.com/YBIFoundation/Dataset/raw/main/Wine.csv')

In [3]:
#3
wine.describe()

class_label     alcohol  malic_acid         ash  alcalinity_of_ash  \
count   178.000000  178.000000  178.000000  178.000000         178.000000   
mean      1.938202   13.000618    2.336348    2.366517          19.494944   
std       0.775035    0.811827    1.117146    0.274344           3.339564   
min       1.000000   11.030000    0.740000    1.360000          10.600000   
25%       1.000000   12.362500    1.602500    2.210000          17.200000   
50%       2.000000   13.050000    1.865000    2.360000          19.500000   
75%       3.000000   13.677500    3.082500    2.557500          21.500000   
max       3.000000   14.830000    5.800000    3.230000          30.000000   

        magnesium  total_phenols  flavanoids  nonflavanoid_phenols  \
count  178.000000     178.000000  178.000000            178.000000   
mean    99.741573       2.295112    2.029270              0.361854   
std     14.282484       0.625851    0.998859              0.124453   
min     70.000000       0.980000    0.340000              0.130000   
25%     88.000000       1.742500    1.205000              0.270000   
50%     98.000000       2.355000    2.135000              0.340000   
75%    107.000000       2.800000    2.875000              0.437500   
max    162.000000       3.880000    5.080000              0.660000   

       proanthocyanins  color_intensity         hue       od280      proline  
count       178.000000       178.000000  178.000000  178.000000   178.000000  
mean          1.590899         5.058090    0.957449    2.611685   746.893258  
std           0.572359         2.318286    0.228572    0.709990   314.907474  
min           0.410000         1.280000    0.480000    1.270000   278.000000  
25%           1.250000         3.220000    0.782500    1.937500   500.500000  
50%           1.555000         4.690000    0.965000    2.780000   673.500000  
75%           1.950000         6.200000    1.120000    3.170000   985.000000  
max           3.580000        13.000000    1.710000    4.000000  1680.000000

In [4]:
#3
wine.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 178 entries, 0 to 177
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   class_label           178 non-null    int64  
 1   class_name            178 non-null    object 
 2   alcohol               178 non-null    float64
 3   malic_acid            178 non-null    float64
 4   ash                   178 non-null    float64
 5   alcalinity_of_ash     178 non-null    float64
 6   magnesium             178 non-null    int64  
 7   total_phenols         178 non-null    float64
 8   flavanoids            178 non-null    float64
 9   nonflavanoid_phenols  178 non-null    float64
 10  proanthocyanins       178 non-null    float64
 11  color_intensity       178 non-null    float64
 12  hue                   178 non-null    float64
 13  od280                 178 non-null    float64
 14  proline               178 non-null    int64  
dtypes: float64(11), int64(3

In [5]:
#3
wine.shape

(178, 15)

In [6]:
wine.columns

Index(['class_label', 'class_name', 'alcohol', 'malic_acid', 'ash',
       'alcalinity_of_ash', 'magnesium', 'total_phenols', 'flavanoids',
       'nonflavanoid_phenols', 'proanthocyanins', 'color_intensity', 'hue',
       'od280', 'proline'],
      dtype='object')

In [7]:
y = wine['class_name']

In [8]:
X = wine[['color_intensity','proline', 'alcohol', 'malic_acid', 'ash',
       'alcalinity_of_ash', 'magnesium', 'total_phenols', 'flavanoids',
       'nonflavanoid_phenols', 'proanthocyanins', 'hue',
       'od280']]

In [9]:
#4
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, train_size=0.7, random_state=2529)

In [10]:
#5
from sklearn.linear_model import LogisticRegression
model = LogisticRegression(max_iter=5000)

In [11]:
#6
model.fit(X_train,y_train)

LogisticRegression(max_iter=5000)

In [12]:
model.intercept_

array([ -4.99474125, -16.3636514 ,  21.35839265])

In [13]:
#7
y_pred = model.predict(X_test)

In [14]:
y_pred

array(['Barbera', 'Barbera', 'Grignolino', 'Barolo', 'Grignolino',
       'Grignolino', 'Barolo', 'Barbera', 'Barolo', 'Grignolino',
       'Grignolino', 'Grignolino', 'Barbera', 'Barolo', 'Barbera',
       'Barolo', 'Barbera', 'Barbera', 'Barolo', 'Grignolino', 'Barbera',
       'Grignolino', 'Barolo', 'Barbera', 'Barbera', 'Grignolino',
       'Grignolino', 'Barbera', 'Barolo', 'Grignolino', 'Grignolino',
       'Barolo', 'Grignolino', 'Barolo', 'Grignolino', 'Grignolino',
       'Barolo', 'Barbera', 'Grignolino', 'Barolo', 'Barolo', 'Barolo',
       'Barolo', 'Barbera', 'Grignolino', 'Barolo', 'Barbera',
       'Grignolino', 'Barolo', 'Grignolino', 'Barolo', 'Barbera',
       'Barbera', 'Barbera'], dtype=object)

In [15]:
#8
from sklearn.metrics import accuracy_score
accuracy_score(y_test,y_pred)

0.9629629629629629

I selected Logistic Regression model because one of the input is categorical variable and i took,
Just color_intensity which gave me 77% accuracy.
then i used Random forest which gave 69% accuracy.
and other then i involved two more metrics as input from dataset that gave me 92%.
finally  used all other metrics except class lable which game me 96% accuracy. finally to predict 3 wine's brand from the metrics.